lib/rubocop/cop/style/array_intersect.rb

```diff
@@ -0,0 +1,99 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module Style
+      # In Ruby 3.1, `Array#intersect?` has been added.
+      #
+      # This cop identifies places where `(receiver & argument).any?`
+      # can be replaced by `receiver.intersect?(argument)`.
+      #
+      # The `receiver.intersect?(argument)` method is faster than
+      # `(receiver & argument).any?` and is more readable.
```

Comment:
A comment about `AllCops:ActiveSupportExtensionsEnabled` would help users understand. e.g.:

```ruby
# `(receiver & argument).blank?` and `(receiver & argument).present?` are detected
# if `AllCops:ActiveSupportExtensionsEnabled` is `true`.
```

Reply:
Fixed in discussion below.